presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -125,6 +138,14 @@ public Block readBlock(ColumnDescriptor columnDescriptor, Type type)
         return columnReader.readBlock(type);
     }
 
+    private byte[] allocateBlock(int length)
+    {
+        byte[] buffer = new byte[length];
+        LocalMemoryContext blockMemoryContext = currentRowGroupMemoryContext.newLocalMemoryContext();
+        blockMemoryContext.setBytes(buffer.length);
```

Comment:
This doesn't look correct.  The `setBytes` method should be setting the currently used memory of the whole context, so each time you allocate a new block you it clears the previous allocation.  Also, the buffer appears to be a temporary variable and not "held" by the reader, so it shouldn't be declared as system memory.